modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java

```diff
@@ -1093,17 +1096,35 @@ private IgniteCheckedException shutdown(Collection<CacheStoreHolder> holders, bo
      * @return Aggregating exception, if error occurred.
      */
     private IgniteCheckedException shutdown(CacheStoreHolder holder, boolean cleanFile,
-        @Nullable IgniteCheckedException aggr) {
+        @Nullable IgniteCheckedException aggr
+    ) {
         aggr = shutdown(holder.idxStore, cleanFile, aggr);
 
         for (PageStore store : holder.partStores) {
             if (store != null)
                 aggr = shutdown(store, cleanFile, aggr);
         }
 
+        if (cleanFile)
+            cleanParentDirectories(holder);
+
         return aggr;
     }
 
+    private void cleanParentDirectories(CacheStoreHolder holder) {
+        Arrays.stream(holder.partStores).map(store -> store.getPath().getParent()).findAny().ifPresent(dir -> {
+            if (Files.isDirectory(dir, LinkOption.NOFOLLOW_LINKS)) {
+                try {
+                    if (!Files.list(dir).findFirst().isPresent())
```

Comment:
I see, you only remove the directory if it's not empty. What happens to `cache_data.dat` file, do we remove it when we delete cache? I'm not sure I know where that happens.

Reply:
Yes, because `cache_data.dat` is removed a little bit earlier as a part of entire cache decommission process here: https://github.com/gridgain/gridgain/blob/c942a4f899b5228a6c8084dfe58f34f9de6afdc7/modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java#L1166-L1166